policy_test.go

```diff
@@ -720,34 +773,61 @@ func TestComputeDynamicPolicy(t *testing.T) {
 				MockPolicyPCRParam{PCR: 12, Alg: tpm2.HashAlgorithmSHA256, Digests: tpm2.DigestList{makePCRDigestFromEvents(tpm2.HashAlgorithmSHA256, "bar")}}},
-			policyCount: 10,
-			policy:      decodeHexString(t, "c84b1520993dc6688afacc423394c7e4e7f1b85a02a520ed0dbafdc81f501faa"),
+			policyCount:  10,
+			pcrSelection: tpm2.PCRSelectionList{{Hash: tpm2.HashAlgorithmSHA256, Select: []int{7, 12}}},
+			policy:       decodeHexString(t, "3cbe37896850d15904508ddf7a28f776642fe60e10b8c9b35e22f50bdc3a53dc"),
+		},
+		{
+			desc:    "MultiplePCRAlgorithms",
+			alg:     tpm2.HashAlgorithmSHA256,
+			signAlg: tpm2.HashAlgorithmSHA256,
+			pcrParams: []MockPolicyPCRParam{
+				MockPolicyPCRParam{PCR: 7, Alg: tpm2.HashAlgorithmSHA512, Digests: tpm2.DigestList{makePCRDigestFromEvents(tpm2.HashAlgorithmSHA256, "foo")}},
+				MockPolicyPCRParam{PCR: 8, Alg: tpm2.HashAlgorithmSHA256, Digests: tpm2.DigestList{makePCRDigestFromEvents(tpm2.HashAlgorithmSHA256, "bar")}}},
+			policyCount:  10,
+			pcrSelection: tpm2.PCRSelectionList{{Hash: tpm2.HashAlgorithmSHA512, Select: []int{7}}, {Hash: tpm2.HashAlgorithmSHA256, Select: []int{8}}},
+			policy:       decodeHexString(t, "1744562fbc67243d2a85fc293d9ccba1381ee835890044b23af957d0be44486e"),
 		},
 	} {
 		t.Run(data.desc, func(t *testing.T) {
 			dataout, err := ComputeDynamicPolicy(data.alg, NewDynamicPolicyComputeParams(key, data.signAlg, data.pcrParams, pinName, data.policyCount))
 			if err != nil {
 				t.Fatalf("ComputeDynamicPolicy failed; %v", err)
 			}
-			if len(dataout.PCRData) != len(data.pcrParams) {
-				t.Fatalf("Unexpected number of PCR data entries")
-			}
-			for i, p := range dataout.PCRData {
-				if p.PCR != data.pcrParams[i].PCR {
-					t.Errorf("Unexpected PCR index")
-				}
-				if p.Alg != data.pcrParams[i].Alg {
-					t.Errorf("Unexpected PCR algorithm")
-				}
-				if len(p.OrDigests) != len(data.pcrParams[i].Digests) {
-					t.Errorf("Unexpected number of OR digests")
-				}
-				for _, d := range p.OrDigests {
-					if len(d) != data.alg.Size() {
-						t.Errorf("Unexpected OR digest size")
+			if len(dataout.PCRSelection) != len(data.pcrSelection) {
+				t.Errorf("Unexpected PCR selection length")
+			} else {
+				for i := range dataout.PCRSelection {
+					if dataout.PCRSelection[i].Hash != data.pcrSelection[i].Hash {
+						t.Errorf("Unexpected algorithm for PCR selection")
 					}
+					if len(dataout.PCRSelection[i].Select) != len(data.pcrSelection[i].Select) {
+						t.Errorf("Unexpected number of PCRs in selection")
+					} else {
+						for j := range dataout.PCRSelection[i].Select {
+							if dataout.PCRSelection[i].Select[j] != data.pcrSelection[i].Select[j] {
+								t.Errorf("Unexpected PCR in selection")
+							}
+						}
+					}
+				}
+			}
+			numPcrOrDigests := 0
+			for _, p := range data.pcrParams {
+				if numPcrOrDigests == 0 {
+					numPcrOrDigests = 1
 				}
+				numPcrOrDigests *= len(p.Digests)
 			}
+			//if len(dataout.PCROrDigests) != numPcrOrDigests {
```

Comment:
Why is this commented out? Should there be a comment about the why?

Reply:
It should have been deleted - I've got another branch that makes additional changes to this code and removes this commented out block already.